---
title: 編譯器錯誤 CS1015
ms.date: 07/20/2015
f1_keywords:
- CS1015
helpviewer_keywords:
- CS1015
ms.assetid: 53179feb-e8be-41e0-bb0b-f7879e9fa613
ms.openlocfilehash: c1a94c6d15737986cd8d203e66d2828e02eefa95
ms.sourcegitcommit: 10986410e59ff29f2ec55c6759bde3eb4d1a00cb
ms.translationtype: MT
ms.contentlocale: zh-TW
ms.lasthandoff: 05/31/2019
ms.locfileid: "66423147"
---
# <a name="compiler-error-cs1015"></a><span data-ttu-id="c92f8-102">編譯器錯誤 CS1015</span><span class="sxs-lookup"><span data-stu-id="c92f8-102">Compiler Error CS1015</span></span>
<span data-ttu-id="c92f8-103">必須是物件、字串或類別類型。</span><span class="sxs-lookup"><span data-stu-id="c92f8-103">An object, string, or class type expected</span></span>  
  
 <span data-ttu-id="c92f8-104">已嘗試將預先定義的資料類型傳遞到 [catch](../../csharp/language-reference/keywords/try-catch.md) 區塊。</span><span class="sxs-lookup"><span data-stu-id="c92f8-104">An attempt was made to pass a predefined data type into a [catch](../../csharp/language-reference/keywords/try-catch.md) block.</span></span> <span data-ttu-id="c92f8-105">只有衍生自 <xref:System.Exception?displayProperty=nameWithType> 的資料類型才能傳遞至 `catch` 區塊。</span><span class="sxs-lookup"><span data-stu-id="c92f8-105">Only data types that derive from <xref:System.Exception?displayProperty=nameWithType> can be passed into a `catch` block.</span></span> <span data-ttu-id="c92f8-106">如需有關例外狀況的詳細資訊，請參閱[例外狀況和例外狀況處理](../../csharp/programming-guide/exceptions/index.md)。</span><span class="sxs-lookup"><span data-stu-id="c92f8-106">For more information on exceptions, see [Exceptions and Exception Handling](../../csharp/programming-guide/exceptions/index.md).</span></span>  
  
## <a name="example"></a><span data-ttu-id="c92f8-107">範例</span><span class="sxs-lookup"><span data-stu-id="c92f8-107">Example</span></span>  
 <span data-ttu-id="c92f8-108">下列範例會產生 CS1015：</span><span class="sxs-lookup"><span data-stu-id="c92f8-108">The following sample generates CS1015:</span></span>  
  
```csharp  
// CS1015.cs  
class Sample  
{  
    static void Main()  
    {  
        try   
        {  
        }  
        catch(int)   // CS1015, int is not derived from System.Exception  
        {  
        }  
    }  
}  
```
